sys/fido2/ctap/ctap_utils.c

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (C) 2021 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup fido2_ctap_utils
+ * @{
+ * @file
+ *
+ * @author     Nils Ollrogge <nils.ollrogge@fu-berlin.de>
+ * @}
+ */
+
+#include <string.h>
+
+#include "xtimer.h"
+
+#include "fido2/ctap.h"
+#include "fido2/ctap/ctap_utils.h"
+
+#if !IS_ACTIVE(CONFIG_FIDO2_CTAP_DISABLE_UP)
+#include "periph/gpio.h"
+#endif
+
+#define ENABLE_DEBUG    (0)
+#include "debug.h"
+
+#if !IS_ACTIVE(CONFIG_FIDO2_CTAP_DISABLE_UP)
+/**
+ * @brief Flag holding information if user is present or not
+ */
+static bool _user_present = false;
+
+/**
+ * @brief Button callback function
+ */
+static void _gpio_cb(void *arg);
+
+int fido2_ctap_utils_init_gpio_pin(void)
+{
+    if (gpio_init_int(BTN0_PIN, BTN0_MODE, GPIO_FALLING, _gpio_cb, NULL) < 0) {
```

Comment:
Why hardcode `BTN0_PIN` when you could just as well have

```suggestion
int fido2_ctap_utils_init_gpio_pin(gpio_t pin, gpio_mode_t mode, gpio_flank_t flank)
{
    if (gpio_init_int(pin, mode, flank, _gpio_cb, NULL) < 0) {
```